internal/services/containers/kubernetes_addons.go

```diff
@@ -235,6 +239,28 @@ func schemaKubernetesAddOnProfiles() *pluginsdk.Schema {
 						},
 					},
 				},
+				"azure_keyvault_secrets_provider": {
+					Type:     pluginsdk.TypeList,
+					MaxItems: 1,
+					Optional: true,
+					Elem: &pluginsdk.Resource{
+						Schema: map[string]*pluginsdk.Schema{
+							"enabled": {
+								Type:     pluginsdk.TypeBool,
+								Required: true,
+							},
```

Comment:
@tombuildsstuff, @stephybun asked a similar question earlier: https://github.com/hashicorp/terraform-provider-azurerm/pull/14308#discussion_r756155324

This wouldn't match with the other addon blocks which have their own `enabled` status. Plus I worry if we have existing clusters that have it manually enabled but the Terraform isn't updated, it may try to disable them on subsequent applies.

Reply:
I just saw your below comment about consistency with other blocks. I'm still worried how this will play out as more addons are created and manually applied during previews. I realize that with a diff or preview, we should catch these possible changes, but if it is being automated it may not be.